lib/index.js

```diff
@@ -0,0 +1,187 @@
+bedrock.events.on('bedrock-express.configure.routes', app => {
+  const routes = config['authn-did-jwt'].routes;
+
+  app.post(routes.login, api.login);
+
+  // only add route for generating tokens if there is a key store
+  if(strategy.store) {
+    app.post(routes.token, optionallyAuthenticated, (req, res, next) => {
+      if(req.user && 'identity' in req.user) {
+        const payload = {
+          sub: req.user.identity.id
+        };
+        if(req.body.aud) {
+          if(req.body.aud.indexOf('https:') !== 0) {
+            return next(new BedrockError(
+              '`aud` must be an HTTPS URL.', 'InvalidAud',
+              {aud: req.body.aud, public: true, httpStatusCode: 400}));
+          }
+          payload.aud = req.body.aud;
+        }
+        return api.generateToken(payload, (err, token) => {
+          if(err) {
+            return next(err);
+          }
+          res.json({token: token});
+        });
+      }
+      const v = validate('authn-did-jwt.postToken', req.body);
+      if(!v.valid) {
+        return next(v.error);
+      }
+      const email = req.body.email.toLowerCase();
+      async.auto({
+        getIdentity: callback => {
+          brIdentity.getAll(
+            null, {'identity.email': email}, {
+              'identity.id': true,
+              'identity.email': true
+            }, (err, result) => {
+              if(err) {
+                return callback(err);
+              }
+              if(result.length === 0) {
+                return callback(new BedrockError(
+                  'Identity not found.', 'NotFound',
+                  {email: email, public: true, httpStatusCode: 404}));
+              }
+              callback(null, result[0].identity);
+            });
+        },
+        generateToken: ['getIdentity', (callback, results) =>{
+          const did = results.getIdentity.id;
+          api.generateToken({sub: did}, callback);
+        }],
+        emit: ['generateToken', (callback, results) => {
+          const event = {
+            type: 'bedrock.authn-did-jwt.token',
+            identity: results.getIdentity,
+            token: results.generateToken
+          };
+          bedrock.events.emitLater(event);
+          callback();
+        }]
+      }, err => {
+        if(err) {
+          return next(err);
+        }
+        res.sendStatus(204);
+      });
+    }); // end routes.token
+  }
+});  // end routes
+
+/**
+ * Generate a JWT.
+ *
+ * @param payload the payload to encode into the JWT.
+ * @param callback callback(err, token) called when done.
+ */
+api.generateToken = function(payload, callback) {
+  if(!strategy.store) {
+    return callback(new BedrockError(
+      'No key store has been set.', 'NotFound'));
+  }
+  // TODO: `sub` could also be an HTTP URL in the future
+  if(!payload.sub || !(/^did:/.test(payload.sub))) {
+    return callback(new BedrockError(
+      '`options.payload.sub` must be a decentralized identifier.',
+      'InvalidPayload'));
```

Comment:
Hmm, trying to think of a better, more generate error code than `InvalidPayload`. We can leave it for now, but we should try to come use more generic error codes like `SyntaxError` or `NotSupported`.